src/vdom/index.js

```diff
@@ -10,19 +12,38 @@ export function isSameNodeType(node, vnode, hydrating) {
 	if (typeof vnode==='string' || typeof vnode==='number') {
 		return node.splitText!==undefined;
 	}
-	if (typeof vnode.nodeName==='string') {
-		return !node._componentConstructor && isNamedNode(node, vnode.nodeName);
+	const { _componentConstructor } = node;
```

Comment:
These were referenced in several places, so I just saved them as constants so they get reused instead of accessing the object several times.